### ISSUE TITLE
Bug 58124170: [React Native New Architecture -> Modal]: Name property and control type is not defined for all the 'Open Modal' buttons.

### ISSUE THREAD

**Issue by vineethkuttan** [microsoft/react-native-gallery#582]
**Build Details:**
**OS Version**: Dev (OS Build 27881.1000)
**App**: React Native Gallery Experimental
**AT**: Narrator

 

**Repro-Steps**:
Launch the React Native Gallery- Experimental app.
Navigate to the 'Expand menu' button and invoke it.
Navigate to 'Home' button and invoke it.
Navigate to 'Modal' button and invoke it.
Navigate to all the 'Open Modal' button.
Observe the announcement and behavior.
**Actual Result:**
Name property and control type is not defined for all the 'Open Modal' buttons. On navigating Narrator only announces as "Group".

**Expected Result:**
Name property should be defined as 'Open Modal' and control type should be defined as 'button'. And on navigating to it, Narrator should announce as "Open Modal, button".

**User impact:**

Users with visual impairment who rely on screen reader will face difficulties if Narrator does not announce the name property for buttons.

MAS Reference:  [MAS 4.1.2 – Name, Role, Value](https://aka.ms/MAS4.1.2)

Trap ID: [1.4 - Uncomprehended Element](https://aka.ms/UncomprehendedElement)
